packages/astro/src/runtime/server/astro-island.ts

```diff
@@ -43,7 +25,14 @@ declare const Astro: {
 			class extends HTMLElement {
 				public Component: any;
 				public hydrator: any;
-				static observedAttributes = ['props'];
+
+				public opts: Record<string, any> = {};
```

Comment:
Would be nice to put a comment above here that these properties are coming from the `<script>` child.